packages/changed-elements-react/src/widgets/comparisonJobWidget/hooks/useNamedVersionLoader.tsx

```diff
@@ -0,0 +1,344 @@
+        const jobStatusAndJobProgress: JobStatusAndJobProgress = await getJobStatusAndJobProgress(args.comparisonJobClient, entry, args.iTwinId, args.iModelId, currentVersionId);
+        entry = {
+          version: entry.version,
+          state: VersionProcessedState.Processed,
+          jobStatus: jobStatusAndJobProgress.jobStatus,
+          jobProgress: jobStatusAndJobProgress.jobProgress,
+        };
+        idEntryMap.set(entry.version.id, entry);
+      }
+      entries = [...idEntryMap.values()];
+      updatingEntries = entries.filter((entry) => entry.jobStatus === "Processing" || entry.jobStatus === "Queued");
+      args.namedVersionLoaderState.result = {
+        namedVersions: { currentVersion: args.namedVersionLoaderState.result.namedVersions.currentVersion, entries: entries },
+      };
+      args.setResult(args.namedVersionLoaderState.result);
+    }
+  }
+};
+
+const getJobStatusAndJobProgress = async (comparisonJobClient: IComparisonJobClient, entry: Entry, iTwinId: string, iModelId: string, currentChangesetId: string): Promise<JobStatusAndJobProgress> => {
+  try {
+    const res = await comparisonJobClient.getComparisonJob({
+      iTwinId: iTwinId,
+      iModelId: iModelId,
+      jobId: `${entry.version.changesetId}-${currentChangesetId}`,
+    });
+    if (res) {
+      switch (res.comparisonJob.status) {
+        case "Completed": {
+          return {
+            jobStatus: "Available",
+            jobProgress: {
+              currentProgress: 0,
+              maxProgress: 0,
+            },
+          };
+        }
+        case "Queued": {
+          return {
+            jobStatus: "Queued",
+            jobProgress: {
+              currentProgress: 0,
+              maxProgress: 0,
+            },
+          };
+        }
+        case "Started": {
+          const progressingJob = res as ComparisonJobStarted;
+          return {
+            jobStatus: "Processing",
+            jobProgress: {
+              currentProgress: progressingJob.comparisonJob.currentProgress,
+              maxProgress: progressingJob.comparisonJob.maxProgress,
+            },
+          };
+        }
+        case "Error":
+          return {
+            jobStatus: "Error",
+            jobProgress: {
+              currentProgress: 0,
+              maxProgress: 0,
+            },
+          };
+      }
+    }
+    return {
+      jobStatus: "Unknown",
+      jobProgress: {
+        currentProgress: 0,
+        maxProgress: 0,
+      },
+    };
+  } catch (_) {
+    return {
+      jobStatus: "Not Processed",
+      jobProgress: {
+        currentProgress: 0,
+        maxProgress: 0,
+      },
+    };
+  }
+};
+
+/**
+ * Used for debugging.
+ * Deletes comparison job.
+ * @throws if delete is not successful
+ */
+//  const deleteJob = async (comparisonJobClient: IComparisonJobClient, entry: Entry, iTwinId: string, iModelId: string, currentChangesetId: string): Promise<void> => {
```

Comment:
I think it's OK to expose the delete job as a function from the package via the client if that's already not done. I would rather this comment be removed or made public.

Reply:
I should remove them, but I kept them for now in case I have to demo again. It has been making it easier to show things off. I'll remove it on my last push.

delete job is already exposed in the client. this is just code to clean up jobs to make getting ready for demo easier.